regression/synthetic_dataset.py

```diff
@@ -1,200 +1,306 @@
-        config=config, retry_strategy=oci.retry.DEFAULT_RETRY_STRATEGY)
-    namespace = oci_client.get_namespace().data
-    objects = oci_client.list_objects(namespace, obj.netloc, prefix=obj.path.lstrip('/'))
+    Args:
+        kwargs (Dict[str, str]): Command-line arguments.
 
-    for filenames in objects.data.objects:
-        oci_client.delete_object(namespace, obj.netloc, filenames.name)
+    Returns:
+        Dict[str, Any]: Dataset parameters.
+    """
+    dataset_params = {}
+    if 'num_samples' in kwargs:
+        dataset_params['num_samples'] = int(kwargs['num_samples'])
+    if 'seed' in kwargs:
+        dataset_params['seed'] = int(kwargs['seed'])
+    if 'shape' in kwargs:
+        dataset_params['shape'] = ast.literal_eval(kwargs['shape'])
+    if 'column_names' in kwargs:
+        dataset_params['column_names'] = ast.literal_eval(kwargs['column_names'])
+    return dataset_params
 
 
-def main(args: Namespace) -> None:
-    """Benchmark time taken to generate the epoch for a given dataset.
+def main(args: Namespace, kwargs: Dict[str, str]) -> None:
+    """Create and delete a dataset.
 
     Args:
-        args (Namespace): Command-line arguments.
+        args (Namespace): Arguments.
+        kwargs (Dict[str, str]): Named arguments.
     """
-    remote_dir = args.cloud_url if args.cloud_url is not None else utils.get_local_remote_dir()
+    dataset_params = get_dataset_params(kwargs)
+    writer_params = get_writer_params(kwargs)
+    if args.name.lower() not in DATASET_MAP:
+        raise ValueError(f'Unsupported dataset {args.name}. Supported: {DATASET_MAP.keys()}')
+    dataset = DATASET_MAP[args.name.lower()](**dataset_params)
+    columns = {name: dtype for name, dtype in zip(dataset.column_names, dataset.column_encodings)}
+
     if args.create:
-        dataset = get_dataset(args.num_samples)
-        with MDSWriter(
-                out=remote_dir,
-                columns=_COLUMNS,
-                compression=args.compression,
-                hashes=args.hashes,
-                size_limit=args.size_limit,
-        ) as out:
+        with MDSWriter(out=args.out, columns=columns, **writer_params) as out:
             for sample in dataset:
                 out.write(sample)
-    if args.delete:
-        obj = urllib.parse.urlparse(remote_dir)
+    elif args.delete:
+        shutil.rmtree(args.out, ignore_errors=True)
+        obj = urllib.parse.urlparse(args.out)
         cloud = obj.scheme
         if cloud == '':
-            shutil.rmtree(remote_dir, ignore_errors=True)
+            shutil.rmtree(args.out, ignore_errors=True)
         elif cloud == 'gs':
-            delete_gcs(remote_dir)
+            delete_gcs(args.out)
         elif cloud == 's3':
-            delete_s3(remote_dir)
+            delete_s3(args.out)
         elif cloud == 'oci':
-            delete_oci(remote_dir)
+            delete_oci(args.out)
 
 
 if __name__ == '__main__':
-    main(parse_args())
+    DATASET_MAP = {
```

Comment:
Should this be at the top of the file prepended with _?

Reply:
I mean as a private global variable

Reply:
Done in https://github.com/mosaicml/streaming/pull/358/commits/74253df1aeba4e68fb0603dac5b0cc90472c3d06